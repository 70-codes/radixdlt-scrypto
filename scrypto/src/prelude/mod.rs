//=============
// This crate
//=============

pub use crate::component::*;
pub use crate::engine::*;
pub use crate::modules::*;
pub use crate::resource::*;
pub use crate::runtime::*;
pub use crate::{
    blueprint, debug, define_permissions, error, external_blueprint, external_component,
    include_code, include_schema, info, main_permissions, metadata, metadata_init,
    method_permission, method_permissions, method_royalties, module_permissions, resource_list, royalties,
    this_package, trace, warn, NonFungibleData, define_roles, role_definition_entry
};

//=========================
// Radix Engine Interface
//=========================

pub use num_traits::{
    cast::FromPrimitive, cast::ToPrimitive, identities::One, identities::Zero, pow::Pow,
    sign::Signed,
};
pub use radix_engine_interface::blueprints::resource::*;
pub use radix_engine_interface::constants::*;
pub use radix_engine_interface::crypto::*;
pub use radix_engine_interface::data::manifest::model::*;
pub use radix_engine_interface::data::manifest::*;
pub use radix_engine_interface::data::scrypto::model::*;
pub use radix_engine_interface::data::scrypto::*;
pub use radix_engine_interface::math::*;
pub use radix_engine_interface::time::*;
pub use radix_engine_interface::traits::*;
pub use radix_engine_interface::types::*;
pub use radix_engine_interface::{
    access_and_or, access_rule_node, dec, i, manifest_args, pdec, role_entry, roles, rule,
    scrypto_args, ScryptoCategorize, ScryptoDecode, ScryptoEncode, ScryptoEvent, ScryptoSbor,
};

//=======
// SBOR
//=======

pub use sbor::rust::prelude::*;
pub use sbor::{Categorize, Decode, DecodeError, Encode, Sbor};

// Needed for macros
pub use radix_engine_interface::radix_engine_common;
