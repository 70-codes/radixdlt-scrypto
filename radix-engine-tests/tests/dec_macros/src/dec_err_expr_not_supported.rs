use radix_engine_interface::prelude::*;

fn main() {
    // Invalid expression
    const X: Decimal = dec!(a);
    let _ = format!("{:?}", X);
}
