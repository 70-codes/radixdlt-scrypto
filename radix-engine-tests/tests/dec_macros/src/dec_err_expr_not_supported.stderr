error: This macro only supports string, integer and float literals.
 --> src/fail.rs:4:29
  |
4 |     const X: Decimal = dec!(a);
  |                             ^
